monash-human-power/MHP_DAS
# Middle Wireless Module sensor readings

## Description

The PR adds classes to read from the MQ135 (co2 concentration), DHT22 (temperature and humidity values) and MPU6050 (accelerometer and gyroscope values) sensors attached to the middle wireless module. It also adds code to collect all the data from each of these sensors and send them over the internet using MQTT. The classes should also be reusable for the other wireless modules.

After this PR the only software component of the middle wireless module left will be reading the battery percentage level.

## Screenshots

This is the middle wireless module sending data and the `test.py` file receiving and displaying those data sets:
<img width="1008" alt="Screen Shot 2020-09-04 at 2 01 24 pm" src="https://user-images.githubusercontent.com/63642262/92198710-6f7bd100-eeb8-11ea-92eb-9e66d7a218ef.png">


## Steps to Test

1) Upload the files to your ESP32 by running the command `./upload.sh <port_name>"` and replacing the port name placeholder with the name of your ESP32 device.
2) Run the `main.py` file using either ampy (`ampy -p <port_name> -b 115200 run main.py`), rshell or picocom (`picocom /dev/tty.SLAB_USBtoUART -b 115200`)
3) Run the `test.py` file inside the middle_module folder or 
4) use Mosquitto from the terminal to publish to the start (`mosquitto_pub -h broker_name -t /v3/wireless-module/2/start -m ""`) and stop (`mosquitto_pub -h broker_name -t /v3/wireless-module/2/start -m ""`) message and subscribe to the data topic manually (`mosquitto_sub -h broker_name -t /v3/wireless-module/2/data`).

Note: It might be a bit hard to test the program without the sensors, but if you have any of the three sensors mentioned above you could connect it to your ESP32 and modify the `main.py` file to remove the other sensors and connect your one, it should be quite straightforward. Otherwise, you could use a publicly hosted MQTT server, such as "broker.hivemq.com" and let me know when you want to test the sensor values and I can power up the wireless module I have :) This way you won't necessarily get to play around with the sensor and check the values but at least get some sense of the workings.
